common/src/main/java/org/opensearch/ml/common/dataset/remote/RemoteInferenceInputDataSet.java

```diff
@@ -46,6 +59,12 @@ public void writeTo(StreamOutput streamOutput) throws IOException {
         } else {
             streamOutput.writeBoolean(false);
         }
+        if (actionType != null) {
+            streamOutput.writeBoolean(true);
```

Comment:
we might need BWC check here.

Reply:
good call out. added that in the latest commit.